aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -60,39 +65,40 @@ use move_core_types::{
     ident_str,
     identifier::Identifier,
     language_storage::{ModuleId, TypeTag},
+    move_resource::MoveStructType,
     transaction_argument::convert_txn_args,
     value::{serialize_values, MoveValue},
     vm_status::StatusType,
 };
-use move_vm_runtime::session::SerializedReturnValues;
+use move_vm_runtime::{logging::expect_no_verification_errors, session::SerializedReturnValues};
 use move_vm_types::gas::UnmeteredGasMeter;
 use num_cpus;
 use once_cell::sync::{Lazy, OnceCell};
 use std::{
+    borrow::Borrow,
     cmp::min,
     collections::{BTreeMap, BTreeSet},
-    convert::{AsMut, AsRef},
     marker::Sync,
     sync::{
         atomic::{AtomicBool, Ordering},
         Arc,
     },
 };
 
+const MAXIMUM_APPROVED_TRANSACTION_SIZE: u64 = 1024 * 1024;
+
 static EXECUTION_CONCURRENCY_LEVEL: OnceCell<usize> = OnceCell::new();
 static NUM_PROOF_READING_THREADS: OnceCell<usize> = OnceCell::new();
 static PARANOID_TYPE_CHECKS: OnceCell<bool> = OnceCell::new();
 static PROCESSED_TRANSACTIONS_DETAILED_COUNTERS: OnceCell<bool> = OnceCell::new();
 static TIMED_FEATURE_OVERRIDE: OnceCell<TimedFeatureOverride> = OnceCell::new();
 
-pub static RAYON_EXEC_POOL: Lazy<Arc<rayon::ThreadPool>> = Lazy::new(|| {
-    Arc::new(
-        rayon::ThreadPoolBuilder::new()
-            .num_threads(num_cpus::get())
-            .thread_name(|index| format!("par_exec_{}", index))
-            .build()
-            .unwrap(),
-    )
+static RAYON_EXEC_POOL: Lazy<rayon::ThreadPool> = Lazy::new(|| {
```

Comment:
The arc change looks great, but I see a different potential issue, let's discuss here but doesn't need to be addressed here.

@sitalkedia @zekun000 @danielxiangzl 
Having the RAYON_EXEC_POOL here seems dangerous to me (sorry that I couldn't review the original PR that moved it in time) - the problem is, the way our current Block-STM implementation uses it, with long-running workers that may sleep, it's almost guaranteed that we can and will deadlock if someone uses the threadpool concurrently with block execution. And here we receive all the callbacks from the Move-VM, so it's super easy to make that mistake.

I think we can move it to block_executor/mod.rs maybe, but not here..